### PR TITLE
Added bandit to CI to avoid security issues

### PR DESCRIPTION
**Pull request by thedrow**
[Bandit](https://github.com/PyCQA/bandit) is a static analysis tool that finds security issues within Python code.
Since we deal with authorization we should avoid insecure code.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -27,3 +27,10 @@ whitelist_externals=echo
 commands=
         python setup.py check -r -s
         echo setup.py/long description is syntaxly correct
+
+[testenv:bandit]
+basepython=python3.7
```
Comment (skion) [inline tox.ini]: Perhaps remove this so that it also runs on machines w/o Python 3.7.

Comment (thedrow) [reply tox.ini]: So it would pick whatever python version available?

Comment (skion) [reply tox.ini]: That was my thinking indeed...

File: tox.ini [hunk 2]
```diff
@@ -27,3 +27,10 @@ whitelist_externals=echo
...
+skipsdist=True
+deps=bandit
+commands=bandit -b bandit.json -r oauthlib/
+whitelist_externals=bandit
```
[the remark this answers is not in the document]
Comment (thedrow) [reply tox.ini]: It refused to run on my machine.

[the remark this answers is not in the document]
Comment (JonathanHuot) [reply tox.ini]: Ok, let's keep it then. I guess `whilelist_externals` become relevant when `bandit` is installed system-wide and its PATH takes priority over the local virtualenv ?

File: bandit.json [hunk 3]
```diff
@@ -0,0 +1,830 @@
+{
+  "errors": [],
+  "generated_at": "2018-12-13T10:39:37Z",
+  "metrics": {
```
[the remark this answers is not in the document]
Comment (thedrow) [reply bandit.json]: Done.

**Pull request merged**